src/app/child-dev-project/children/child-details/form-components/form-subcomponent.ts

```diff
@@ -0,0 +1,91 @@
+import { OnChanges, SimpleChanges } from "@angular/core";
+import { Child } from "../../model/child";
+import { AbstractControlOptions, FormBuilder, FormGroup } from "@angular/forms";
+import { EntityMapperService } from "../../../../core/entity/entity-mapper.service";
+import { AlertService } from "../../../../core/alerts/alert.service";
+
+export abstract class FormSubcomponent implements OnChanges {
+  child: Child = new Child("");
+  editing: boolean = false;
+  form: FormGroup;
+  validateForm: boolean = false;
+
+  public constructor(
+    private entityMapperService: EntityMapperService,
+    private fb: FormBuilder,
+    private alertService: AlertService
+  ) {
+    this.initForm(this.getFormConfig());
+  }
+
+  protected abstract getFormConfig(): {
+    controlsConfig: { [key: string]: any };
```

Comment:
can we type these more specifically than `any`?

Reply:
Its the type definition for the arguments of the `fb.group(...)`. I dont think we can make it any more specific